lib/selections.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @typedef {Object} MinimalSelectionItem
+ * @property {number} from
+ * @property {number} to
+ */
+
+/** A selection of pieces to download.
+ * @typedef {MinimalSelectionItem & {
+ *  offset: number,
+ *  priority: number,
+ *  notify: function,
+ *  isStreamSelection?: boolean
+ * }} SelectionItem
+ */
+
+export class Selections {
+  /**
+   * @type {Array<SelectionItem>}
+   */
+  _items = []
+
+  /**
+   * @param {MinimalSelectionItem & {isStreamSelection: boolean, priority?: number}} item Interval to be removed from the selection
+   */
+  remove (item) {
+    for (let i = 0; i < this._items.length; i++) {
```

Comment:
should this be a `for (const existing of this) {` then later use either existing.remove() or this.swap()?

Reply:
nvm it wouldn't be swap, but remove could be used

Reply:
we also need `i` for the `splice` call on line 55.